libc/string_errsig.c

```diff
@@ -0,0 +1,247 @@
+{
+	errno = 0;
+	TEST_ASSERT_NOT_NULL(strerror(0));
+	TEST_ASSERT_LESS_OR_EQUAL_INT(MAX_LEN_STRING, strlen(strerror(0)));
+	TEST_ASSERT_EQUAL_INT(0, errno);
+}
+
+
+TEST(string_errsingn, strerror_invaild)
+{
+	int values[] = { INT_MIN, INT_MIN / 2, -1024, 1024, INT_MAX / 2, INT_MAX };
+	for (int i = 0; i < (sizeof values / sizeof *values); i++) {
+		TEST_ASSERT_NOT_NULL(strerror(values[i]));
+		TEST_ASSERT_NOT_NULL(strstr(strerror(values[i]), "Unknown error"));
+		TEST_ASSERT_LESS_OR_EQUAL_INT(MAX_LEN_STRING, strlen(strerror(values[i])));
+	}
+}
+
+
+TEST(string_errsingn, strerror_r_basic)
+{
+/* strerror_r() not yet implemented in Phoenix-RTOS */
+#ifdef __phoenix__
+	TEST_IGNORE();
+#else
+	for (int i = 0; i < (sizeof error_codes / sizeof *error_codes); i++) {
+		char prev_buf[MAX_LEN_STRING] = { 0 };
+		char buf[MAX_LEN_STRING] = { 0 };
+
+		errno = 0;
+		TEST_ASSERT_EQUAL_INT(0, strerror_r(error_codes[i - 1], prev_buf, sizeof(buf)));
+		TEST_ASSERT_EQUAL_INT(0, strerror_r(error_codes[i], buf, sizeof(buf)));
+		TEST_ASSERT_LESS_OR_EQUAL_INT(MAX_LEN_STRING, strlen(prev_buf));
+		TEST_ASSERT_LESS_OR_EQUAL_INT(MAX_LEN_STRING, strlen(buf));
+		TEST_ASSERT_TRUE(buf != prev_buf);
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+#endif
+}
+
+
+TEST(string_errsingn, strerror_r_zero)
+{
+/* strerror_r() not yet implemented in Phoenix-RTOS */
+#ifdef __phoenix__
+	TEST_IGNORE();
+#else
+	char buf[MAX_LEN_STRING];
+	errno = 0;
+	TEST_ASSERT_EQUAL_INT(0, strerror_r(0, buf, sizeof(buf)));
+	TEST_ASSERT_LESS_OR_EQUAL_INT(MAX_LEN_STRING, strlen(buf));
+	TEST_ASSERT_EQUAL_INT(0, errno);
+#endif
+}
+
+
+TEST(string_errsingn, strerror_r_invaild)
+{
+/* strerror_r() not yet implemented in Phoenix-RTOS */
+#ifdef __phoenix__
+	TEST_IGNORE();
+#else
+	int values[] = { INT_MIN, INT_MIN / 2, -1024, 1024, INT_MAX / 2, INT_MAX };
+	for (int i = 0; i < (sizeof values / sizeof *values); i++) {
+		char buf[MAX_LEN_STRING] = { 0 };
+		errno = 0;
+		TEST_ASSERT_EQUAL_INT(EINVAL, strerror_r(values[i], buf, sizeof(buf)));
+		TEST_ASSERT_LESS_OR_EQUAL_INT(MAX_LEN_STRING, strlen(buf));
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+#endif
+}
+
+
+TEST(string_errsingn, strerror_r_overflow)
+{
+/* strerror_r() not yet implemented in Phoenix-RTOS */
+#ifdef __phoenix__
+	TEST_IGNORE();
+#else
+	char buf[1];
+	for (int i = 0; i < (sizeof error_codes / sizeof *error_codes); i++) {
+		errno = 0;
+		TEST_ASSERT_EQUAL_INT(ERANGE, strerror_r(error_codes[i], buf, sizeof(buf)));
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+#endif
+}
+
+
+TEST(string_errsingn, strsignal_basic)
+{
+/* Implemented in Phoenix-RTOS, but not required in POSIX*/
+#ifdef __phoenix__
+	TEST_ASSERT_EQUAL_STRING("Information request", strsignal(SIGINFO));
+	TEST_ASSERT_EQUAL_STRING("EMT trap", strsignal(SIGEMT));
+#endif
+
+/* Disabled because of #NUMBER issue */
+#ifndef __phoenix__
+	TEST_ASSERT_EQUAL_STRING("I/O possible", strsignal(SIGPOLL));
+#endif
+
+	TEST_ASSERT_EQUAL_STRING("Hangup", strsignal(SIGHUP));
```

Comment:
nah, don't test the strings - they are locale-depenant and are not portable!

eg. with `LANG=pl_PL.UTF-8` this returns `Rozłączenie`